explainer.md

```diff
@@ -257,57 +257,60 @@ The semantics defined in each apply only to their associated custom element inst
 
 ```js
 class CustomTab extends HTMLElement {
+  #internals = null;
+  #tablist = null;
+  #tabpanel = null;
+
   constructor() {
     super();
-
-    let _privates = {};
+    this.#internals = customElements.createInternals(this);
+    this.#internals.role = "tab";
   }
 
   // Observe the custom "active" attribute.
   static get observedAttributes() { return ["active"]; }
 
-  createdCallback(elementInternals) {
-    // Keep a private reference to the internals object.
-    _privates.internals = elementInternals;
-  }
-
   connectedCallback() {
-    _privates.tablist = this.parentElement;
+    this.#tablist = this.parentElement;
   }
 
   setTabPanel(tabpanel) {
     if (tabpanel.localName !== "custom-tabpanel" || tabPanel.id === "")
       return;  // fail silently
 
-    _privates.tabpanel = tabpanel;
+    this.#tabpanel = tabpanel;
     tabpanel.setTab(this);
-    _privates.internals.ariaControls = tabPanel;    // does not reflect
+    this.#internals.ariaControls = tabPanel;    // does not reflect
   }
 
   // ... setters/getters for custom properties which reflect to attributes
 
-  // When the custom "active" attribute changes,
-  // keep the accessible checked state in sync.
   attributeChangedCallback(name, oldValue, newValue) {
     switch(name) {
-    case "active":
-      let active = (newValue != null);
-      _privates.tabpanel.shown = active;
-      _privates.internals.ariaSelected = (newValue !== null);
-      if (selected)
-        this._tablist.setSelectedTab(this);  // ensure no other tab has "active" set
+      case "active":
+        let active = (newValue != null);
+        this.#tabpanel.shown = active;
+
+        // When the custom "active" attribute changes,
+        // keep the accessible "selected" state in sync.
+        this.#internals.ariaSelected = (newValue !== null);
+
+        if (selected)
+          this.#tablist.setSelectedTab(this);  // ensure no other tab has "active" set
         break;
     }
   }
 }
 
-customElements.define("custom-tab", CustomTab, { role: "tab" });
+customElements.define("custom-tab", CustomTab, { role: "tab", needsElementInternals: true });
```

Comment:
I don't think we really need `needsElementInternals` here. Custom elements should be allowed to get the internals if they please to independently of the registration process.